crates/wit-component/src/lib.rs

```diff
@@ -85,7 +85,13 @@ pub(crate) fn base_producers() -> wasm_metadata::Producers {
 
 /// Parse a WIT file from a path that represents a top level 'wit' directory,
 /// normally containing a 'deps' folder.
-pub fn parse_wit_from_path(path: impl AsRef<Path>) -> Result<(Resolve, PackageId)> {
+#[cfg(feature = "wat")]
```

Comment:
It feels like this function should continue to be defined without the `wat` feature and instead `is_wasm_binary_or_wat` should _only_ check for the binary preamble when the `wat` feature is disabled?

Reply:
Oh and we'd obviously skip the `wat::parse_bytes` call.

Not worth the complexity?

Reply:
Nah makes sense to me!